File: Python/Easy/144.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []

        # preorder traversal is current left then right

        answer = []

        # current
        answer.append(root.val)

        # left
        answer.extend(self.preorderTraversal(root.left))

        # right
        answer.extend(self.preorderTraversal(root.right))

        return answer

File: Java/Easy/144.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        // base case
        if (root == null) {
            return new ArrayList<>();
        }

        // preorder is current then left then right
        ArrayList<Integer> answer = new ArrayList<>();

        // current
        answer.add(root.val);

        // left
        answer.addAll(preorderTraversal(root.left));

        // right
        answer.addAll(preorderTraversal(root.right));

        return answer;

    }
}